server/src/main/java/io/crate/planner/node/ddl/AlterTableAddColumnPlan.java

```diff
@@ -185,7 +185,12 @@ static void addExistingPrimaryKeys(DocTableInfo tableInfo, AnalyzedTableElements
             if (reference.valueType().id() != ObjectType.ID) {
                 columnDef.indexConstraint(reference.indexType());
             }
-            columnDef.dataType(reference.valueType().getName());
+
+            // We are mirroring PK type as is (including type's internal settings if there are any)
+            // We cannot resolve type by name via columnDef.dataType(reference.valueType().getName())
+            // as internal parameters, such as VARCHAR length can be lost.
+            columnDef.dataType(reference.valueType());
```

Comment:
I'm not sure whether we cn have issues with other usages of
` public void dataType(String dataType)` or  
`public void dataType(String typeName, List<Integer> parameters, boolean logWarnings)`

if it's about not using same instance maybe just introduce safe, internal state aware `clone` for all types? another option could be just checking all such cases, currently only `bit` and `varchar`, `numeric` is not stored - but I find it error prone.

Reply:
I think these here would be worth double checking: 

https://github.com/crate/crate/blob/06ba1def3ea6db6793ae98188f435c82ea9cb35a/server/src/main/java/io/crate/analyze/AnalyzedTableElements.java#L407-L412

The other usages look safe to me.

Reply:
I checked in https://github.com/crate/crate/pull/12510/commits/d6a81fa3097102c8d1d00736b63d74f2845c5d04 and found another bug.  I will prepare separate commit with own fix entry/test case + rewrite test to a planner test

I was wondering whether I should just remove error prone `public void dataType(String dataType)` and use passed type instance everywhere and in places where we use constant type?

For example, instead of `context.analyzedColumnDefinition.dataType("string")` use 
`context.analyzedColumnDefinition.dataType(StringType.Instance)`?

Reply:
Would be possible, but I think the root cause of the problem is that we need to add existing columns here at all.

Ideally a ALTER TABLE operation would only require the delta to be present in the request, and everything else would happen in the cluster state update action. This is partly related to https://github.com/crate/crate/issues/11939, https://github.com/crate/crate/issues/10025 and what I'm currently looking into cleaning up

Reply:
fixed another issue, updated changes and replaced integration test with planner test in https://github.com/crate/crate/pull/12510/commits/5b918b98d75656bef355d0cf6fe29e51e14bff60